src/v/kafka/server/quota_manager.h

```diff
@@ -54,35 +55,22 @@ class quota_manager : public ss::peering_sharded_service<quota_manager> {
     using clock = ss::lowres_clock;
 
     struct throttle_delay {
-        bool first_violation;
+        bool enforce = false;
         clock::duration duration;
+        clock::duration enforce_duration() const {
+            if (enforce) {
+                return duration;
+            } else {
+                return clock::duration::zero();
```

Comment:
+1. I was thinking about that too and came to the same conclusion.

Reply:
>+1. I was thinking about that too and came to the same conclusion.

yeh, i only looked because i was curious if seastar optimized for already expired timers. i'm kinda surprised that they dont' have that optimization.

Reply:
>i think that it still makes sense to avoid calling sleep_abortable when throttling is not enforced. even though the semantics are exactly what you want, sleep_abortable unconditionally does a heap allocation, links up the abort source back references, creates a promise, and schedules it.

it's not clear if this happened or not. is there now a relationship in which enforce implies a positive duration?